src/core/highlight-vars.js

```diff
@@ -0,0 +1,93 @@
+// Module core/highlight-vars
+// Highlights occurrences of a <var> within a section on click
+// Set `conf.highlightVars = true` to enable
+// removes highlights from <var> if clicked anywhere else
+// all is done while keeping in mind that exported html stays clean (removeHighlight)
+
+import { sub } from "core/pubsubhub";
+import hlVars from "deps/text!core/css/var.css";
+
+export const name = "core/highlight-vars";
+
+export function run(conf) {
+  if (!conf.highlightVars) {
+    return;
+  }
+  const styleElement = document.createElement("style");
+  styleElement.textContent = hlVars;
+  styleElement.classList.add("removeOnSave");
+  document.head.appendChild(styleElement);
+
+  document.addEventListener("click", initHighlight);
+
+  // remove highlights, cleanup empty class/style attributes
+  sub("beforesave", outputDoc => {
+    [...outputDoc.querySelectorAll("var.respec-active")].forEach(
+      removeHighlight
+    );
+  });
+}
+
+// availability of highlight colors.
+const HL_COLORS = new Map([
+  // [ background:String, available:Boolean ]
+  // or [ background,color:String, available:Boolean ]
+  ["yellow", true],
+  ["#726012,#fff", true],
+  ["#c78e00", true],
+  ["#3949ab,#fff", true],
+  ["#F720E5", true],
+]);
+
+function getHighlightColor(target) {
+  // return current colors if applicable
+  const bg = target.style.getPropertyValue("--respec-background-color");
+  const col = target.style.getPropertyValue("--respec-color");
+  if (bg) return `${bg}${col ? `,${col}` : ""}`;
+
+  // first color preference
+  if (HL_COLORS.get("yellow") === true) return "yellow";
+
+  // otherwise get some other available color
+  return [...HL_COLORS.keys()].find(c => HL_COLORS.get(c) === true) || "yellow";
+}
+
+function removeHighlight(el) {
+  // done so that only the respec classes are removed
+  el.classList.remove("respec-active");
+  el.style.removeProperty("--respec-background-color");
+  el.style.removeProperty("--respec-color");
+  // clean up empty class/style attributes so they don't come in export
+  if (!el.getAttribute("style")) el.removeAttribute("style");
+  if (!el.getAttribute("class")) el.removeAttribute("class");
+}
+
+function highlightVars(target) {
+  const textContent = target.textContent.trim();
+  const parent = target.closest("section");
+  const highlightColor = getHighlightColor(target);
+  [...parent.querySelectorAll("var")]
+    .filter(el => el.textContent.trim() === textContent)
+    .forEach(highlightVar);
+}
+
+function highlightVar(elem) {
```

Comment:
@sidvishnoi, I moved this out, but broke it in the process 👍. However, this means something is not right with the design of the algorithm. It seems to have too many side effects. Can you look at refactoring this?

Reply:
I made a little fix here: https://github.com/sidvishnoi/respec/commit/c1f263c0b4bb8f710137963bb3803261dd8d7ebe

Reply:
Also, do have a look at this approach https://github.com/sidvishnoi/respec/commit/4154f8bdeb917d92d550e2fd7bd4f2dfc8cf9988

Did this because there was a common repetitive work going in `highlightVar` to set color status and splitting color to `[background, color]`. This comes at an expense of initialisation of a new variable `varsToHighlight` which II have found a little expensive.

Reply:
👹 says "Premature optimization is the root of all evil". But seriously, maintainability trumps any expense of initialization here.

Reply:
If I understood it right, you meant the second one is more maintainable.
So I made another commit https://github.com/sidvishnoi/respec/commit/07c229054d4bb5380704a33b18e33028bd9f0c37 to separate out `addHighlight` and `removeHighlight` logic.

You might want to merge the [sidvishnoi/ISSUE_1529-alt](https://github.com/sidvishnoi/respec/tree/ISSUE_1529-alt/) branch into `w3c/sidvishnoi-ISSUE_1529` if you think it's right.

Reply:
I think what's in sidvishnoi/respec@07c2290 should be ok.

Reply:
@sidvishnoi, I think we are 99% done here. Can you bring in sidvishnoi/respec@07c2290?